### ISSUE TITLE
PT FSDP support

### ISSUE THREAD

**Issue by albertz** [rwth-i6/returnn#1736]
RETURNN should have direct support for FSDP.

I think we should just follow the [FSDP tutorial](https://docs.pytorch.org/tutorials/intermediate/FSDP_tutorial.html).

We also should run some benchmarks on how speed compares to our existing distributed training approaches, but taking advantage of each method (e.g. I think for a given model size, FSDP should allow larger batch sizes).

(cc @NeoLegends @dthulke @robin-p-schmitt @curufinwe)
